### DISCUSSION TITLE
Invalid encoding proof generated at layer 1

### DISCUSSION THREAD

**Discussion by ssim-itaz** [filecoin-project/lotus#8713] [General Q&A]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [X] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.15.3-dev+mainnet+git.33a05d33a+api1.5.0
Local: lotus-miner version 1.15.3-dev+mainnet+git.33a05d33a
```

### Describe the Bug

thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread 'thread '<unnamed><unnamed>' panicked at '' panicked at 'Invalid encoding proof generated at layer 1Invalid encoding proof generated at layer 1', ', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs/root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs::261261:thread ':33<unnamed>33
' panicked at '
Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread 'thread '<unnamed><unnamed>' panicked at '' panicked at 'Invalid encoding proof generated at layer 1Invalid encoding proof generated at layer 1', ', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs/root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs::261261::3333

thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:thread '261<unnamed>:' panicked at '33Invalid encoding proof generated at layer 1
', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
thread '<unnamed>' panicked at 'Invalid encoding proof generated at layer 1', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-11.0.2/src/stacked/vanilla/proof.rs:261:33
2022-05-24T06:09:55.468Z	WARN	ffiwrapper	ffiwrapper/sealer_cgo.go:740	checking PreCommit failed: Rust panic: no unwind information
2022-05-24T06:09:55.468Z	WARN	ffiwrapper	ffiwrapper/sealer_cgo.go:741	num:19919 tkt:[87 118 126 54 45 98 10 33 189 190 90 84 193 39 165 242 34 73 79 46 75 43 19 75 1 37 102 163 198 242 206 59] seed:[242 138 103 247 171 117 198 0 42 206 130 147 56 150 134 5 254 121 246 145 155 58 141 95 196 174 208 87 208 174 3 65] sealedCID:bagboea4b5abcarzhw2w2mbrsuvatnojb7r4lwg5hgax2y5ceung2hik5rodcc23n, unsealedCID:baga6ea4seaqao7s73y24kcutaosvacpdjgfe5pw76ooefnyqw4ynr3d2y6x2mpq

### Logging Information

```text
precommit2 error
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by rjan90** [comment]
Hey @ssim-itaz! Since `Rust panic: no unwind information` is usually caused by bad graphics driver or bad ram. I am moving this to the Github discussion so we can investigate more. If we find out that there is an issue with Lotus we will move it back into the `issue` section.

I have a couple of questions:
What kind of Nvidia-driver are you using?
And are you using OpenCL or CUDA?
Can you run a Memtest? And are you using ECC RAM?

**Comment by ssim-itaz** [comment]
What kind of Nvidia-driver are you using?
-> nvidia-driver-510,  cuda-toolkit-11-7, cuda-drivers-510
And are you using OpenCL or CUDA?
-> using CUDA
Can you run a Memtest? And are you using ECC RAM?
-> Yes memory, ssd, gpu, cpu test OK
and using ECC-RD RAM

**Reply by rjan90** [reply]
Okay, good! I see that you also are running the master-branch, can you please revert to a more stable release like [v1.15.2](https://github.com/filecoin-project/lotus/releases/tag/v1.15.2) or [v1.15.3-rc1](https://github.com/filecoin-project/lotus/releases/tag/v1.15.3-rc1).

**Reply by ssim-itaz** [reply]
I will test it after changing to version 1.15.2

**Reply by rjan90** [reply]
Nice 👍

**Reply by ssim-itaz** [reply]
Even after installing '1.15.2', the same error occurred. 
I reinstalled the 'OS, 1.15.3, CUDA, RUST, GO'. 'No problem.

**Comment by ssim-itaz** [comment]
I think the 'rust version' is the problem.
My server install 'rust version' is 1.61, and the version downloaded when compiling lotus is '1.58'.

**Comment by beck-8** [comment]
The log is more detailed,`lotus-miner sectors status --log xx`
There are only three reasons for the problem
- parent_cache is error
- pc2 tasks was killed at running tree r 
- gpu driver or gpu hardware has problem

Trust me, you will find the answer
